Files/UserControls/Search/SizeRangeSlider.xaml.cs

```diff
@@ -0,0 +1,132 @@
+
+        private readonly IList<Size> Steps = new List<Size>
+        {
+            new Size(0),
+            new Size(1),
+            new Size(256),
+            new Size(1, Size.Units.Kibi),
+            new Size(16, Size.Units.Kibi),
+            new Size(256, Size.Units.Kibi),
+            new Size(1, Size.Units.Mebi),
+            new Size(5, Size.Units.Mebi),
+            new Size(16, Size.Units.Mebi),
+            new Size(64, Size.Units.Mebi),
+            new Size(128, Size.Units.Mebi),
+            new Size(256, Size.Units.Mebi),
+            new Size(512, Size.Units.Mebi),
+            new Size(1, Size.Units.Gibi),
+            new Size(5, Size.Units.Gibi),
+            new Size(16, Size.Units.Gibi),
+            new Size(64, Size.Units.Gibi),
+            Size.MaxValue,
+        };
+
+        public static readonly DependencyProperty RangeProperty =
+            DependencyProperty.Register(nameof(Range), typeof(SizeRange), typeof(SizeRangeSlider), new PropertyMetadata(null));
+
+        private static readonly DependencyProperty ToolTipProperty =
+            DependencyProperty.Register(nameof(ToolTip), typeof(string), typeof(SizeRangeSlider), new PropertyMetadata(null));
+
+        public SizeRange Range
+        {
+            get => (SizeRange)GetValue(RangeProperty);
+            set
+            {
+                if (Range.Equals(value))
+                {
+                    return;
+                }
+
+                SetValue(RangeProperty, value);
+
+                var (minSize, maxSize) = (value.MinSize, value.MaxSize);
+
+                int minStep = Steps.Contains(minSize) ? 3 * Steps.IndexOf(minSize) : 3 * Steps.IndexOf(Steps.Last(step => step <= minSize)) + 1;
+                int maxStep = Steps.Contains(maxSize) ? 3 * Steps.IndexOf(maxSize) : 3 * Steps.IndexOf(Steps.Last(step => step <= maxSize)) + 2;
+
+                isInUpdateRange = true;
+                if (Selector.RangeStart > maxStep)
+                {
+                    Selector.RangeStart = minStep;
+                    Selector.RangeEnd = maxStep;
+                }
+                else
+                {
+                    Selector.RangeEnd = maxStep;
+                    Selector.RangeStart = minStep;
+                }
+                isInUpdateRange = false;
+            }
+        }
+
+        private string ToolTip
+        {
+            get => (string)GetValue(ToolTipProperty);
+            set
+            {
+                SetValue(ToolTipProperty, value);
+                if (ToolTipBlock is not null)
+                {
+                    int index = int.Parse(value);
+                    var size = (index % 3) switch
+                    {
+                        1 => Range.MinSize,
+                        2 => Range.MaxSize,
+                        _ => Steps[index / 3],
+                    };
+                    ToolTipBlock.Text = size.ToString("N");
+
+                    // show fake tooltip inside popup.
```

Comment:
What's this for?

Reply:
Size implements IFormattable to provide many formatting options. It seems simpler and more scalable to me than having many properties or a function with an enum parameter. The downside is that it is not very readable. 

This returns the named label: the size with unit or "No Limit" for the maximum value.

Reply:
But what's the fake tooltip for?

Reply:
It is about a tooltip which is displayed when one manipulates one of the 2 cursors (min or max value). Using a real tooltip is not possible because it is displayed too late and it is not practical while moving a cursor. RangeSelector developers have therefore made their own tooltip. I took again their template, which I modified to display a modified value in the tooltip which only allowed to display a number, and to modify the appearance.

One problem with the fake tooltip is that it does not display out of control, and therefore does not display fully when on edge. So I added placement code to correct this. I had planned to propose this modification directly in the source project, so as not to have to manage it ourselves.

Reply:
Thanks for the explanation :+1: